src/Ractive/prototype/set.js

```diff
@@ -10,9 +10,9 @@ export default function Ractive$set ( keypath, value ) {
 	if ( isObject( keypath ) ) {
 		const map = keypath;
 
-		for ( keypath in map ) {
-			if ( map.hasOwnProperty( keypath) ) {
-				set( this, keypath, map[ keypath ] );
+		for ( const k in map ) {
+			if ( map.hasOwnProperty( k) ) {
+				set( this, k, map[k] );
 			}
```

Comment:
Huh – seems the fastest might actually be a combo, as you avoid both the function overhead and the cost of calling `hasOwnProperty`:

```js
const keypaths = Object.keys( map );
let i = keypaths.length;
while ( i-- ) {
  set( this, keypaths[i], map[ keypaths[i] ] );
}
```

Via https://jsperf.com/object-keys-vs-for-in-with-closure/3